homeassistant/components/vesync/config_flow.py

```diff
@@ -0,0 +1,78 @@
+"""Config flow utilities."""
+import logging
+from collections import OrderedDict
+import voluptuous as vol
+from homeassistant import config_entries
+from homeassistant.helpers import config_validation as cv
+from homeassistant.core import callback
+from homeassistant.const import (CONF_TIME_ZONE, CONF_USERNAME,
+                                 CONF_PASSWORD)
+
+DOMAIN = 'vesync'
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@callback
+def configured_instances(hass):
+    """Return already configured instances."""
+    return [
+        entry.data[CONF_USERNAME]
+        for entry in hass.config_entries.async_entries(DOMAIN)]
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class VeSyncFlowHandler(config_entries.ConfigFlow):
+    """Handle a config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    def __init__(self):
+        """Instantiate config flow."""
+        self._username = None
+        self._password = None
+        self._time_zone = None
+        self.data_schema = OrderedDict()
+        self.data_schema[vol.Required(CONF_USERNAME)] = str
+        self.data_schema[vol.Required(CONF_PASSWORD)] = str
+        self.data_schema[vol.Optional(CONF_TIME_ZONE)] = str
```

Comment:
Why do we need to set timezone? Isn't home assistant core config already aware  of that?

Reply:
I see in some components they test to see if the time zone is not set.  Is this redundant?

Reply:
Please give me an example and I can look into it.

Reply:
These are the one's I found.  I was just trying to avoid any errors in setting the component up.  If HA won't load without defaulting to a time zone, I will take it out

https://github.com/home-assistant/home-assistant/blob/f195ecca4b410461cec94664da2f903d99beae46/homeassistant/components/tod/binary_sensor.py#L42

https://github.com/home-assistant/home-assistant/blob/f269135ae925de9a0cf8d90750da9a6ff323415e/homeassistant/components/eight_sleep/__init__.py#L90

https://github.com/home-assistant/home-assistant/blob/2b490e44862ca7daae113fff092202ce9a646823/homeassistant/components/time_date/sensor.py#L38

Reply:
I took time zone out completely, I saw balloobs message on chat